### PR TITLE
[9.0][ADD] Sale Stock Medical Prescription US

### PR DESCRIPTION
**Pull request by woodbrettm**
- [x] Depends #89

### CODE REVIEW

File: sale_stock_medical_prescription_us/models/medical_prescription_order_line.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                # @TODO
+                # Should be record.duration_uom_id first, then day_uom_id last
+                # in args
+                day_qty = self.env['product.uom']._compute_qty_obj(
+                    day_uom_id, record.duration, record.duration_uom_id,
+                )
+            else:
+                day_qty = record.duration
+
+            # @TODO
```
Comment (bmw95) [inline sale_stock_medical_prescription_us/models/medical_prescription_order_line.py]: Fix needed: 'What should happen if day_qty is 0? (Line 88 could cause 0 division error)

File: sale_stock_medical_prescription_us/models/medical_prescription_order_line.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            # What should happen if day_qty is 0?
+            if not day_qty:
+                continue
+
+            # @TODO
```
Comment (bmw95) [inline sale_stock_medical_prescription_us/models/medical_prescription_order_line.py]: Fix needed: "record.qty can be set to 0, causing total_qty to be 0 and causing 0 division error in daily_qty

File: sale_stock_medical_prescription_us/models/medical_prescription_order_line.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+            date_first_dispense = fields.Datetime.from_string(
+                first_proc.date_planned,
+            )
+            date_delta = datetime.now() - date_first_dispense
+            days_passed = date_delta.days
+
+            if day_uom_id != record.duration_uom_id:
+
+                # @TODO
```
Comment (woodbrettm) [inline sale_stock_medical_prescription_us/models/medical_prescription_order_line.py]: Fix needed: day_uom_id should be 3rd argument and duration_uom_id first arg. Check medication uom data to see if adjusting there can remedy having to put the code the way it is currently.

File: sale_stock_medical_prescription_us/tests/test_medical_prescription_order_line.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+    def test_compute_refill_qty_remain_active_qty_more_than_qty(self):
+        """ Test refill_qty_remain when active_dispense_qty > qty """
+        self.procurement_1.product_qty = 62
+        self.assertAlmostEqual(
+            self.rx_line_3.refill_qty_remain,
+            2.66,
+            2,
+        )
+
+    # @TODO
```
Comment (woodbrettm) [inline sale_stock_medical_prescription_us/tests/test_medical_prescription_order_line.py]: Test failing from 0 division error

File: sale_stock_medical_prescription_us/models/medical_prescription_order_line.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+                    record.duration,
+                    day_uom_id,
+                )
+            else:
+                day_qty = record.duration
+
+            # @TODO
+            # What should happen if day_qty is 0?
+            if not day_qty:
+                continue
```
Comment (woodbrettm) [inline sale_stock_medical_prescription_us/models/medical_prescription_order_line.py:81]: @lasley

File: sale_stock_medical_prescription_us/models/medical_prescription_order_line.py [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+            # @TODO
+            # record.qty can be set to 0, causing total_qty to be 0
+            # and causing 0 division error for line 86
+            total_qty = record.qty * (record.refill_qty_original + 1.0)
+            daily_qty = total_qty / float(day_qty)
```
Comment (woodbrettm) [inline sale_stock_medical_prescription_us/models/medical_prescription_order_line.py:87]: @lasley

**Comment by woodbrettm** [comment]
Ready for review other than 1 failing test from zero division error and above comments.

**Comment by lasley** [comment]
Closing - we no longer have interest in v9. We will continue these forward in v10.

Anyone is welcome to pick the code up and reintroduce a PR, however.

@bmw95 - please make sure these changes are all in our v10 release branch.

**Pull request closed without merging**